OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -273,5 +280,18 @@ public override IEnumerable<TargetLineNode> TargetLineNodes(Actor self)
 			if (targetLineColor != null)
 				yield return new TargetLineNode(useLastVisibleTarget ? lastVisibleTarget : target, targetLineColor.Value);
 		}
+
+		bool HaveArmamentsFor(Target target)
+		{
+			foreach (var attack in attackTraits)
+			{
+				if (attack.ChooseArmamentsForTarget(target, forceAttack).Any())
+				{
+					return true;
+				}
+			}
```

Comment:
you can remove all the braces here, this is part of our unenforced style guidelines